src/mqtt_client.rs

```diff
@@ -188,6 +187,17 @@ where
         Ok(())
     }
 
+    /// Set custom MQTT port to connect to.
+    ///
+    /// # Note
+    /// This must be completed before connecting to a broker.
+    ///
+    /// # Args
+    /// * `port` - The Port number to connect to.
+    pub fn set_broker_port(&mut self, port: u16) {
+        self.session_state.broker.set_port(port);
```

Comment:
```suggestion
        if (*self.connection_state.state() != States::Restart) 
            return Err(Error::NotReady)'
        }
        
        self.session_state.broker.set_port(port);
```

Reply:
Correct - in the `Restart` state, we don't have any socket allocated yet from the stack. After we get it, we transition to `ConnectTransport` where we call the `TcpClientStack::connect()` command. If we just check for the `Restart` state, we can ensure that we haven't called `connect()` yet.